fuse/host.go

```diff
@@ -708,6 +726,8 @@ func (host *FileSystemHost) Mount(mountpoint string, opts []string) bool {
 // Unmount may be called at any time after the Init() method has been called
 // and before the Destroy() method has been called.
 func (host *FileSystemHost) Unmount() bool {
+	host.RLock()
```

Comment:
The `host.fuse` variable is modified in `hostInit` and `hostDestroy`. The `Unmount` method is documented as being safe only between `Init` and `Destroy`, which means that the file system should provide its own synchronization method to ensure that `Unmount` is not called at the wrong time.

However I agree that it might be better if `Unmount` is improved to catch cases where the file system calls it before `Init` or after `Destroy`.
However your use of the lock here may make things better

Reply:
If I remove the lock here, I get into a race condition when executing `go test -race`; I had the same problem with my fs and I don't know how to fix it on the side of my fs.